processing/src/main/java/org/apache/druid/frame/field/DoubleArrayFieldReader.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.frame.field;
+
+import org.apache.datasketches.memory.Memory;
+import org.apache.druid.segment.ColumnValueSelector;
+
+import javax.annotation.Nullable;
+
+/**
+ * Reader for fields written by {@link NumericArrayFieldWriter#getDoubleArrayFieldWriter}
+ */
+public class DoubleArrayFieldReader extends NumericArrayFieldReader
+{
+  @Override
+  public ColumnValueSelector<?> makeColumnValueSelector(
+      Memory memory,
+      ReadableFieldPointer fieldPointer
+  )
+  {
+    return new Selector<Double>(memory, fieldPointer)
+    {
+
+      @Nullable
+      @Override
+      public Double getIndividualValueAtMemory(Memory memory, long position)
+      {
+        ColumnValueSelector<?> columnValueSelector =
+            DoubleFieldReader.forArray()
+                             .makeColumnValueSelector(memory, new ConstantFieldPointer(position));
```

Comment:
this seems kind of expensive, like we make a column value selector and a constant pointer for every element of every row? surely there is a less garbage intense way to read the array elements, though doesn't necessarily need to be resolved in this PR...

Reply:
Lemme check if I can reduce these object creations post addressing the other review comments.. If it requires a minor refactor, I'll update it here.

Reply:
Refactored, now we are just changing the position, and creating each object exactly once per selector 
(which should probably be once per Frame i think, or slightly higher, if the frame was comprised of multiple memory locations)